### DISCUSSION TITLE
how to use python request call aiflow?

### DISCUSSION THREAD

**Discussion by wy329** [apache/airflow#44467] [Q&A]
I want to use python code call a exist airflow program, like login, trriger dag, activate dag!Is there any exist method? The first step is login with csrf_token stop me. Beg to help me! TIA!

**Comment by wy329** [comment]
Oh!I found https://github.com/apache/airflow-client-python, a offical library!

**Comment by ZhiXIaox** [comment]
import requests

# 定义Aiflow API的端点
api_url = "http://your-aiflow-server/api/endpoint"

# 定义请求头和数据
headers = {
    "Content-Type": "application/json",
    "Authorization": "Bearer YOUR_API_KEY"
}
data = {
    "param1": "value1",
    "param2": "value2"
}

# 发送POST请求
response = requests.post(api_url, headers=headers, json=data)

# 处理响应
if response.status_code == 200:
    result = response.json()
    print("请求成功，结果：", result)
else:
    print("请求失败，状态码：", response.status_code)
